### PR TITLE
style: type checking hack

### PR DESCRIPTION
**Pull request by aarnphm**
A hack to conform with TYPE_CHECKING

Before
Screenshot 2022-10-29 at 20 07 36

After
Screenshot 2022-10-29 at 20 08 24

### CODE REVIEW

**Comment by sauyon** [comment]
Let's call it `ListStr` at least

File: src/bentoml/_internal/bento/build_config.py [hunk 1]
```diff
@@ -284,6 +284,12 @@ def conda_dependencies_validator(
                 )
 
 
+if TYPE_CHECKING:
+    L: t.TypeAlias = list[str]
```
Comment (aarnphm) [inline src/bentoml/_internal/bento/build_config.py:288]:
```suggestion
    ListStr: t.TypeAlias = list[str]
```

File: src/bentoml/_internal/bento/build_config.py [hunk 2]
```diff
@@ -284,6 +284,12 @@ def conda_dependencies_validator(
...
+else:
+    L = list
```
Comment (aarnphm) [inline src/bentoml/_internal/bento/build_config.py:290]:
```suggestion
    ListStr = list
```

File: src/bentoml/_internal/bento/build_config.py [hunk 3]
```diff
@@ -295,14 +301,14 @@ class CondaOptions:
     environment_yml: t.Optional[str] = None
     channels: t.Optional[t.List[str]] = attr.field(
         default=None,
-        validator=attr.validators.optional(attr.validators.instance_of(list)),
+        validator=attr.validators.optional(attr.validators.instance_of(L)),
```
Comment (aarnphm) [inline src/bentoml/_internal/bento/build_config.py:304]:
```suggestion
        validator=attr.validators.optional(attr.validators.instance_of(ListStr)),
```

File: src/bentoml/_internal/bento/build_config.py [hunk 4]
```diff
@@ -295,14 +301,14 @@ class CondaOptions:
...
     )
     dependencies: t.Optional[DependencyType] = attr.field(
         default=None, validator=attr.validators.optional(conda_dependencies_validator)
     )
     pip: t.Optional[t.List[str]] = attr.field(
         default=None,
-        validator=attr.validators.optional(attr.validators.instance_of(list)),
+        validator=attr.validators.optional(attr.validators.instance_of(L)),
```
Comment (aarnphm) [inline src/bentoml/_internal/bento/build_config.py:311]:
```suggestion
        validator=attr.validators.optional(attr.validators.instance_of(ListStr)),
```

File: src/bentoml/_internal/bento/build_config.py [hunk 5]
```diff
@@ -402,7 +408,7 @@ class PythonOptions:
     )
     packages: t.Optional[t.List[str]] = attr.field(
         default=None,
-        validator=attr.validators.optional(attr.validators.instance_of(list)),
+        validator=attr.validators.optional(attr.validators.instance_of(L)),
```
Comment (aarnphm) [inline src/bentoml/_internal/bento/build_config.py:411]:
```suggestion
        validator=attr.validators.optional(attr.validators.instance_of(ListStr)),
```

File: src/bentoml/_internal/bento/build_config.py [hunk 6]
```diff
@@ -418,23 +424,23 @@ class PythonOptions:
     )
     trusted_host: t.Optional[t.List[str]] = attr.field(
         default=None,
-        validator=attr.validators.optional(attr.validators.instance_of(list)),
+        validator=attr.validators.optional(attr.validators.instance_of(L)),
```
Comment (aarnphm) [inline src/bentoml/_internal/bento/build_config.py:427]:
```suggestion
        validator=attr.validators.optional(attr.validators.instance_of(ListStr)),
```

File: src/bentoml/_internal/bento/build_config.py [hunk 7]
```diff
@@ -418,23 +424,23 @@ class PythonOptions:
...
     )
     find_links: t.Optional[t.List[str]] = attr.field(
         default=None,
-        validator=attr.validators.optional(attr.validators.instance_of(list)),
+        validator=attr.validators.optional(attr.validators.instance_of(L)),
```
Comment (aarnphm) [inline src/bentoml/_internal/bento/build_config.py:431]:
```suggestion
        validator=attr.validators.optional(attr.validators.instance_of(ListStr)),
```

File: src/bentoml/_internal/bento/build_config.py [hunk 8]
```diff
@@ -418,23 +424,23 @@ class PythonOptions:
...
     )
     extra_index_url: t.Optional[t.List[str]] = attr.field(
         default=None,
-        validator=attr.validators.optional(attr.validators.instance_of(list)),
+        validator=attr.validators.optional(attr.validators.instance_of(L)),
```
Comment (aarnphm) [inline src/bentoml/_internal/bento/build_config.py:435]:
```suggestion
        validator=attr.validators.optional(attr.validators.instance_of(ListStr)),
```

File: src/bentoml/_internal/bento/build_config.py [hunk 9]
```diff
@@ -418,23 +424,23 @@ class PythonOptions:
...
     )
     pip_args: t.Optional[str] = attr.field(
         default=None,
         validator=attr.validators.optional(attr.validators.instance_of(str)),
     )
     wheels: t.Optional[t.List[str]] = attr.field(
         default=None,
-        validator=attr.validators.optional(attr.validators.instance_of(list)),
+        validator=attr.validators.optional(attr.validators.instance_of(L)),
```
Comment (aarnphm) [inline src/bentoml/_internal/bento/build_config.py:443]:
```suggestion
        validator=attr.validators.optional(attr.validators.instance_of(ListStr)),
```

**Review by sauyon — APPROVED** [review]

**Pull request merged**